docs/performance/tune-performance.md

```diff
@@ -281,6 +281,22 @@ https://github.com/microsoft/onnxruntime/blob/master/docs/python/inference/api_s
 * C#
 https://github.com/microsoft/onnxruntime/blob/master/csharp/test/Microsoft.ML.OnnxRuntime.Tests/OrtIoBindingAllocationTest.cs 
 
+### Convolution heavy models and the CUDA EP
+ORT leverages CuDNN for convolution operations and the first step in this process is to determine which "optimal" convolution algorithm to use while performing the convolution operation for the given input configuration(input shape, filter shape, etc.) in each `Conv` node . This sub-step involves querying CuDNN for a "workspace" memory size and have this allocated so that CuDNN can use this auxiliary memory while determining the "optimal" convolution algorithm to use. By default, ORT clamps the workspace size to 32 MB which may lead to a sub-optimal convolution algorithm getting picked by CuDNN. To allow ORT to allocate the maximum possible workspace as determined by CuDNN, a provider option needs to get set (as shown below). Keep in mind that using this flag may increase the peak memory usage by a factor (sometimes a few GBs) but this does help CuDNN pick the best convolution algorithm for the given input. We have found that this is an important flag to use while using an fp16 model as this allows CuDNN to pick tensor core algorithms for the convolution operations (if the hardware supports tensor core operations). This flag may or may not result in performance gains for other data types (`float` and `double`).
+
+* Python
+```
+sess = ort.InferenceSession("my_conv_heavy_fp16_model.onnx",  sess_options = sess_options, providers=['CUDAExecutionProvider'])
+options = sess.get_provider_options()
+cuda_options = options['CUDAExecutionProvider']
+cuda_options['cudnn_conv_use_max_workspace'] = '1'
+sess.set_providers(['CUDAExecutionProvider'], [cuda_options])
```

Comment:
nit: set_providers will delete the previously created session and re-create it. Would be nice to set these options while constructing InferenceSession so that InferenceSession is constructed only once.

Reply:
I tried it now and it doesn't work - there is a bug somewhere.

Reply:
Did not know that when passing provider options and a provider list to the InferenceSession ctor, we had to send them together in `providers` and that we can't use `providers` and `provider_options` then. Works in the latest iteration.